SU2_CFD/include/variables/CVariable.hpp

```diff
@@ -47,2469 +47,2615 @@
 
   /*!
    * \brief Set the value of the min solution.
-   * \param[in] val_var - Index of the variable.
-   * \param[in] val_solution - Value of the min solution for the index <i>val_var</i>.
+   * \param[in] iPoint - Point index.
+   * \param[in] iVar - Index of the variable.
+   * \param[in] solution - Value of the min solution for the index <i>iVar</i>.
    */
-  inline void SetSolution_Min(unsigned short val_var, su2double val_solution) {Solution_Min[val_var] = val_solution; }
+  inline void SetSolution_Min(Idx_t iPoint, Idx_t iVar, su2double solution) { Solution_Min(iPoint,iVar) = solution; }
 
   /*!
    * \brief Get the value of the slope limiter.
+   * \param[in] iPoint - Point index.
    * \return Pointer to the limiters vector.
    */
-  inline su2double *GetLimiter(void) {return Limiter; }
+  inline su2double *GetLimiter(Idx_t iPoint) { return Limiter[iPoint]; }
 
   /*!
    * \brief Get the value of the slope limiter.
-   * \param[in] val_var - Index of the variable.
-   * \return Value of the limiter vector for the variable <i>val_var</i>.
+   * \param[in] iPoint - Point index.
+   * \param[in] iVar - Index of the variable.
+   * \return Value of the limiter vector for the variable <i>iVar</i>.
    */
-  inline su2double GetLimiter(unsigned short val_var) {return Limiter[val_var]; }
+  inline su2double GetLimiter(Idx_t iPoint, Idx_t iVar) const { return Limiter(iPoint,iVar); }
 
   /*!
    * \brief Get the value of the min solution.
-   * \param[in] val_var - Index of the variable.
-   * \return Value of the min solution for the variable <i>val_var</i>.
+   * \param[in] iPoint - Point index.
+   * \param[in] iVar - Index of the variable.
+   * \return Value of the min solution for the variable <i>iVar</i>.
    */
-  inline su2double GetSolution_Max(unsigned short val_var) {return Solution_Max[val_var]; }
+  inline su2double GetSolution_Max(Idx_t iPoint, Idx_t iVar) const { return Solution_Max(iPoint,iVar); }
 
   /*!
    * \brief Get the value of the min solution.
-   * \param[in] val_var - Index of the variable.
-   * \return Value of the min solution for the variable <i>val_var</i>.
+   * \param[in] iPoint - Point index.
+   * \param[in] iVar - Index of the variable.
+   * \return Value of the min solution for the variable <i>iVar</i>.
    */
-  inline su2double GetSolution_Min(unsigned short val_var) {return Solution_Min[val_var]; }
+  inline su2double GetSolution_Min(Idx_t iPoint, Idx_t iVar) const { return Solution_Min(iPoint,iVar); }
 
-  /*!
-   * \brief Get the value of the preconditioner Beta.
-   * \return Value of the low Mach preconditioner variable Beta
-   */
-  inline virtual su2double GetPreconditioner_Beta() {return 0; }
+//  /*!
+//   * \brief Get the value of the preconditioner Beta.
+//   * \param[in] iPoint - Point index.
+//   * \return Value of the low Mach preconditioner variable Beta
+//   */
+//  inline virtual su2double GetPreconditioner_Beta(Idx_t iPoint) const { return 0.0; }
```

Comment:
These methods for Precond_Beta are only used for the Roe-Turkel scheme, but we only set the value and never get it back.
Also the value is set for iPoint and jPoint based on a result from numerics (which means each value is overwritten multiple times in the edge loop).